src/systems/systems/src/air_conditioning/acs_controller.rs

```diff
@@ -0,0 +1,1808 @@
+        }
+    }
+
+    fn zone_id(&self) -> &str {
+        self.zone_id
+    }
+}
+
+impl DuctTemperature for ZoneController {
+    fn duct_demand_temperature(&self) -> HashMap<&'static str, ThermodynamicTemperature> {
+        let mut demand_temperature: HashMap<&str, ThermodynamicTemperature> = HashMap::new();
+        demand_temperature.insert(self.zone_id, self.duct_demand_temperature);
+        demand_temperature
+    }
+}
+
+impl SimulationElement for ZoneController {
+    fn read(&mut self, reader: &mut SimulatorReader) {
+        self.zone_measured_temperature = reader.read(&self.zone_temp_id);
+    }
+
+    fn write(&self, writer: &mut SimulatorWriter) {
+        // TODO: Replace this with actual duct temperature when mixer is modelled, not duct demand temperature
+        writer.write(&self.zone_duct_temp_id, self.duct_demand_temperature);
+    }
+}
+
+#[derive(Clone, Copy)]
+enum OvhdFlowSelector {
+    Lo = 80,
+    Norm = 100,
+    Hi = 120,
+}
+
+read_write_enum!(OvhdFlowSelector);
+
+impl From<f64> for OvhdFlowSelector {
+    fn from(value: f64) -> Self {
+        match value as u8 {
+            0 => OvhdFlowSelector::Lo,
+            1 => OvhdFlowSelector::Norm,
+            2 => OvhdFlowSelector::Hi,
+            _ => panic!("Overhead flow selector position not recognized."),
+        }
+    }
+}
+
+struct PackFlowController {
+    apu_bleed_valve_open_id: VariableIdentifier,
+    apu_rpm_id: VariableIdentifier,
+    crossbleed_id: VariableIdentifier,
+    ditching_id: VariableIdentifier,
+    engine_1_state_id: VariableIdentifier,
+    engine_2_state_id: VariableIdentifier,
+    eng_1_fire_id: VariableIdentifier,
+    eng_2_fire_id: VariableIdentifier,
+    ovhd_flow_selector_id: VariableIdentifier,
+
+    pack_flow_id: VariableIdentifier,
+
+    flow_demand: Ratio,
+    absolute_flow: MassRate,
+    fcv_1_open_allowed: bool,
+    fcv_2_open_allowed: bool,
+    should_open_fcv: [bool; 2],
+
+    apu_bleed_on: bool,
+    crossbleed_is_on: bool,
+    ditching_is_on: bool,
+    engine_1_in_start_mode: bool,
+    engine_2_in_start_mode: bool,
+    engine_1_on_fire: bool,
+    engine_2_on_fire: bool,
+    flow_control_valve_open: bool,
+    flow_selector_position: OvhdFlowSelector,
+    pack_in_start_condition: bool,
+    single_pack_operation: bool,
+}
+
+impl PackFlowController {
+    const PACK_START_TIME: f64 = 30.;
+    const PACK_START_FLOW_LIMIT: f64 = 100.;
+    const APU_SUPPLY_FLOW_LIMIT: f64 = 120.;
+    const ONE_PACK_FLOW_LIMIT: f64 = 120.;
+    const FLOW_REDUCTION_LIMIT: f64 = 80.;
+    const BACKFLOW_LIMIT: f64 = 80.;
+
+    const FLOW_CONSTANT_C: f64 = 0.5675; // kg/s
+    const FLOW_CONSTANT_XCAB: f64 = 0.00001828; // kg(feet*s)
+
+    fn new(context: &mut InitContext) -> Self {
+        Self {
+            apu_bleed_valve_open_id: context.get_identifier("APU_BLEED_AIR_VALVE_OPEN".to_owned()),
+            apu_rpm_id: context.get_identifier("APU_N_RAW".to_owned()),
```

Comment:
**issue**: Data which is calculated within Rust shouldn't be read through LVars. Instead, pass the necessary information down through `update` calls.

Reply:
Absolutely true. I've replaced all "reads" within `AirConditioningSystemController` by pulling data through the `update` functions within it.